pkg/csi/common.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package csi
+
+import (
+	"fmt"
+
+	snapshotv1beta1api "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/label"
+)
+
+// NewCSIListOptions returns a ListOptions for matching CSI objects based on the Velero backup name and label.
+func NewCSIListOptions(name string) metav1.ListOptions {
+	return metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", velerov1api.BackupNameLabel, label.GetValidName(name)),
```

Comment:
This could potentially be a bug - we're not using the `label.GetValidName` function when we list the objects for serialization. I haven't checked to see if we use it when labeling the objects, but we should use it everywhere in the codebase

Reply:
Yeah, we should probably do an audit there and update wherever needed...IIRC names can be longer than label values, so any time you get a long backup name, it may not be valid as a label value.

Reply:
I'll update the core Velero values in this PR and get another one in for the plugin.

Across the codebase, we'll want to audit and do a comparison for existing objects to see if the label name matches one generated by the function, then update the value.